src/models/ObjectLevelCan.ts

```diff
@@ -1,17 +1,25 @@
-type BasicPermissions = 'create' | 'read' | 'update' | 'delete'
+import { PermissionVerb, WorkspacePermission, workspacePermissions } from '@/services/can'
 
+type PermissionVerbs = PermissionVerb<WorkspacePermission>
+const permissionVerbs = workspacePermissions.map(permission => permission.split(':')[0]) as readonly PermissionVerbs[]
 
-type ObjectLevelCan<TPermissions extends string> = {
-  [key in TPermissions]: boolean
-}
+type ObjectTypesWithPermissions = WorkspacePermission extends `${string}:${infer TObject}` ? TObject : never
+
+type ActionsForObjectsHelper<O extends string, T extends string> = T extends `${infer Action}:${infer TObject}` ? TObject extends O ? Action : never : never
+type PermissionsForObjectType<TObjectType extends ObjectTypesWithPermissions> = ActionsForObjectsHelper<TObjectType, WorkspacePermission>
 
-export type BasicPermissionsObjectLevelCan = ObjectLevelCan<BasicPermissions>
-export type BasicRunnablePermissionsObjectLevelCan = ObjectLevelCan<BasicPermissions | 'run'>
+export type ObjectLevelCan<TObjectType extends ObjectTypesWithPermissions> = {
+  [key in PermissionsForObjectType<TObjectType>]: boolean
+}
 
-export function createObjectLevelCan<T extends Record<string, boolean>>(): T {
-  return new Proxy({} as T, {
-    get() {
-      return true
+export function createObjectLevelCan<T extends ObjectTypesWithPermissions>(): ObjectLevelCan<T> {
+  const knownProperties = permissionVerbs
+  return new Proxy({} as ObjectLevelCan<T>, {
+    get(_target, property) {
+      // only proxy known properties so that vue doesn't think it's a ref in templates
+      if (knownProperties.includes(property as PermissionVerbs)) {
+        return true
+      }
```

Comment:
This is the primary fix - only proxying known properties which is being computed from workspace permissions such that it should constrain the property access proxying to only affect properties like `.create`, `.read`, `.run`, `.update`, and `.delete` along with other verbs if they show up as valid workspace permissions.

The rest of changes are TS-related stuff to just make it a little easier to use this thing typing-wise. Instead of having types for each can type depending on what permission verbs are relevant to that object, TS magic lets you now use `ObjectLevelCan<the name of some object type constrained by real workspace permission values>` and it should _just_ work for knowing that the can should have `.run` for example.